platform_secrets/service.py

```diff
@@ -148,8 +163,26 @@ async def copy_to_namespace(
         if missing_secret_names:
             raise CopyScopeMissingError.create(missing_secret_names)
 
-        # ensure namespace is there
-        await self._kube_client.create_namespace(name=target_namespace)
+        try:
+            # let's try to get a namespace
+            namespace = await self._kube_client.get_namespace(name=target_namespace)
+        except ResourceNotFound:
```

Comment:
I think we should first try to create, then try to fetch. Otherwise in case if two parallel requests one may fail

Reply:
@zubenkoivan great point! fixed